colossalai/inference/struct.py

```diff
@@ -167,15 +170,23 @@ class BatchInfo:
     Information to be passed and used for a batch of sequences.
     """
 
+    max_batch_size: int
+    kv_max_split_num: int
+    num_heads: int
+    head_dim: int
     sequences_set: OrderedSet[Sequence] = None
     is_prompts: bool = True
     device: torch.device = None
+    dtype: torch.dtype = None
+    fd_inter_tensor: FDIntermTensors = None
 
     def __post_init__(self):
         if self.device is None:
             self.device = torch.cuda.current_device()
         if self.sequences_set is None:
             self.sequences_set = OrderedSet()
+        if self.fd_inter_tensor is None:
+            self.fd_inter_tensor = FDIntermTensors()
```

Comment:
Post init seems to be redundant here. You could just set the default values.

Reply:
Done